src/averageAges.js

```diff
@@ -55,7 +71,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let mother;
+  const children = people.filter(kid =>
+    onlyWithSon
+      ? kid.sex === 'm'
+        && people.find(person => person.name === kid.mother)
+      : people.find(person => person.name === kid.mother)
+  );
+
+  return children.reduce((sum, kid) => {
+    mother = people.find(person => person.name === kid.mother);
+
+    return sum + kid.born - mother.born;
+  }, 0) / children.length;
```

Comment:
Good job.
Approve. 
I just want to show example how we can avoid O**2 complexity and solve it just in two iterations
```
 const mothers = people.reduce((acc, person) => {
    acc[person.name] = person;

    return acc;
  }, {});

  const { sum, childrenCount } = people.reduce((acc, person) => {
    const { sex, mother, born } = person;

    const isPerson = onlyWithSon
      ? sex === 'm' && mothers[mother]
      : mothers[mother];

    return isPerson ? {
      sum: acc.sum + born - mothers[mother].born,
      childrenCount: acc.childrenCount + 1,
    } : acc;
  }, {
    sum: 0, childrenCount: 0,
  });

  return sum / childrenCount;
  ```
  of course something like that posiible to make with another functions.